pkg/watch/listwatch/until.go

```diff
@@ -0,0 +1,101 @@
+/*
```

Comment:
Any reason this can't be in client/cache or client/cache/listwatch?  I like the latter because it is defining the role (client), purpose (to cache), and what it does (list watch)